### PR TITLE
storage: mark MergeRequests as "non KV"

### PR DESCRIPTION
**Pull request by petermattis**
Mark MergeRequests as "non KV" so that the skip the command queue and
timestamp cache. Merge requests can be reordered without issue obviating
the need for the command queue. Since there is no MVCC for merge requests
the timestamp cache isn't necessary. In simple local testing the command
queue processing was sometimes taking single-digits of milliseconds.

See #9818

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9889)

### CODE REVIEW

**Comment by petermattis** [comment]
Cc @cockroachdb/stability

**Comment by spencerkimball** [comment]
LGTM.

File: roachpb/api.go [hunk 1]
```diff
@@ -791,7 +791,7 @@ func (*RangeLookupRequest) flags() int        { return isRead }
 func (*ResolveIntentRequest) flags() int      { return isWrite }
 func (*ResolveIntentRangeRequest) flags() int { return isWrite | isRange }
 func (*NoopRequest) flags() int               { return isRead } // slightly special
-func (*MergeRequest) flags() int              { return isWrite }
+func (*MergeRequest) flags() int              { return isWrite | isNonKV }
```
Comment (petermattis) [inline roachpb/api.go]: Done.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**